WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java

```diff
@@ -194,13 +224,30 @@ public void onClick(View view) {
         }
 
         // set the top margin of the title based on whether there's a featured image and post header
-        RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams) postHolder.txtTitle.getLayoutParams();
+        LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) postHolder.txtTitle.getLayoutParams();
         params.topMargin = titleMargin;
 
+        // show word count when appropriate, include reading time if at least two minutes
+        if (post.wordCount > 0 && !post.isDiscoverPost()) {
+            String wordCountStr = String.format(mWordCountFmtStr, post.wordCount);
+            int readingTimeInMinutes = post.wordCount / READING_WORDS_PER_MINUTE;
+            if (readingTimeInMinutes >= MIN_READING_TIME_MINUTES) {
+                wordCountStr += " (~" + String.format(mReadingTimeFmtStr, readingTimeInMinutes) + ")";
+            }
+            postHolder.txtWordCount.setText(wordCountStr);
+            postHolder.txtWordCount.setVisibility(View.VISIBLE);
+        } else {
+            postHolder.txtWordCount.setVisibility(View.GONE);
+        }
+
         // show the best tag for this post
         final String tagToDisplay = (mCurrentTag != null ? post.getTagForDisplay(mCurrentTag.getTagName()) : null);
         if (!TextUtils.isEmpty(tagToDisplay)) {
-            postHolder.txtTag.setText(tagToDisplay);
+            if (tagToDisplay.startsWith("#")) {
```

Comment:
This is the third place I've seen this sort of logic. Maybe worth making a static method for it?